packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -505,3 +510,314 @@ def get_app_metrics(dcos_api_session, node: str, container_id: str):
+
+                # We expect three datapoints, could be in any order
+                uptime_dp = None
+                for dp in app_response.json()['datapoints']:
+                    if dp['name'] == 'statsd_tester.time.uptime':
+                        uptime_dp = dp
+                        break
+
+                # If this metric is missing, statsd-emitter's metrics were not received
+                assert uptime_dp is not None, 'got {}'.format(app_response.json())
+
+                datapoint_keys = ['name', 'value', 'unit', 'timestamp', 'tags']
+                for k in datapoint_keys:
+                    assert k in uptime_dp, 'got {}'.format(uptime_dp)
+
+                expected_tag_names = {
+                    'dcos_cluster_id',
+                    'test_tag_key',
+                    'dcos_cluster_name',
+                    'host'
+                }
+                check_tags(uptime_dp['tags'], expected_tag_names)
+                assert uptime_dp['tags']['test_tag_key'] == 'test_tag_value', 'got {}'.format(uptime_dp)
+
+                assert 'dimensions' in app_response.json(), 'got {}'.format(app_response.json())
+                assert 'task_name' in app_response.json()['dimensions'], 'got {}'.format(
+                    app_response.json()['dimensions'])
+
+                # Look for the specified task name.
+                assert task_name.strip('/') == app_response.json()['dimensions']['task_name'],\
+                    'Nested container was not tagged with the correct task name'
+            else:
+                # Retry for 30 seconds for each parent container to present its
+                # content.
+                @retrying.retry(stop_max_delay=30000)
+                def wait_for_container_response():
+                    response = dcos_api_session.metrics.get(container_id_path, node=agent_ip)
+                    assert response.status_code == 200
+                    return response
+
+                container_response = wait_for_container_response()
+                assert 'datapoints' in container_response.json(), 'got {}'.format(container_response.json())
+
+                cid_registry = []
+                for dp in container_response.json()['datapoints']:
+                    # Verify expected tags are present.
+                    assert 'tags' in dp, 'got {}'.format(dp)
+                    expected_tag_names = {
+                        'container_id',
+                    }
+                    if dp['name'].startswith('blkio.'):
+                        # blkio stats have 'blkio_device' tags.
+                        expected_tag_names.add('blkio_device')
+                    check_tags(dp['tags'], expected_tag_names)
+
+                    # Ensure all container IDs in the response from the
+                    # containers/<id> endpoint are the same.
+                    cid_registry.append(dp['tags']['container_id'])
+                    assert(check_cid(cid_registry))
```

Comment:
If you change `check_cid` as mentioned in my previous comment, then you can move this check outside the loop.